replica/tablets.cc

```diff
@@ -564,15 +564,15 @@ class tablet_incremental_selector : public sstables::incremental_selector_impl {
 
         // pos must be monotonically increasing in the weak sense
         // but caller can skip to a position outside the current set
-        auto token = pos.token();
         if (!_cur_set || pos.token() >= _lowest_next_token) {
+            auto token = find_starting_token(_tset, pos.token());
```

Comment:
Hmm, this violates the way `select` is documented to work:
```
        // Return the sstables that intersect with `pos` and the next
        // position where the intersecting sstables change.
        // To walk through the token range incrementally call `select()`
        // with `dht::ring_position_view::min()` and then pass back the
        // returned `next_position` on each next call until
        // `next_position` becomes `dht::ring_position::max()`.
```

Meaning that if if these is no matching sstable_set for the given `pos`, we should end up in the `!_cur_set` block below and return an empty result, with the lewest_next_position of the next allocated storage_group.